android/expoview/src/main/java/host/exp/exponent/utils/ExperienceActivityUtils.kt

```diff
@@ -0,0 +1,320 @@
+    }
+  }
+
+  fun updateSoftwareKeyboardLayoutMode(manifest: RawManifest, activity: Activity) {
+    val keyboardLayoutMode = manifest.getAndroidKeyboardLayoutMode() ?: "resize"
+
+    // It's only necessary to set this manually for pan, resize is the default for the activity.
+    if (keyboardLayoutMode == "pan") {
+      activity.window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN)
+    }
+  }
+
+  // region user interface style - light/dark/automatic mode
+  /**
+   * Sets uiMode to according to what is being set in manifest.
+   */
+  fun overrideUiMode(manifest: RawManifest, activity: AppCompatActivity) {
+    val userInterfaceStyle = manifest.getAndroidUserInterfaceStyle() ?: "light"
+    activity.delegate.localNightMode = nightModeFromString(userInterfaceStyle)
+  }
+
+  private fun nightModeFromString(userInterfaceStyle: String?): Int {
+    return if (userInterfaceStyle == null) {
+      AppCompatDelegate.MODE_NIGHT_NO
+    } else when (userInterfaceStyle) {
+      "automatic" -> {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) {
+          AppCompatDelegate.MODE_NIGHT_AUTO_BATTERY
+        } else AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
+      }
+      "dark" -> AppCompatDelegate.MODE_NIGHT_YES
+      "light" -> AppCompatDelegate.MODE_NIGHT_NO
+      else -> AppCompatDelegate.MODE_NIGHT_NO
+    }
+  }
+
+  // endregion
+
+  // region StatusBar configuration
+
+  /**
+   * React Native is not using flag [WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS] nor view/manifest attribute 'android:windowTranslucentStatus'
+   * (https://developer.android.com/reference/android/view/WindowManager.LayoutParams.html#FLAG_TRANSLUCENT_STATUS)
+   * (https://developer.android.com/reference/android/R.attr.html#windowTranslucentStatus)
+   * Instead it's using [WindowInsets] to limit available space on the screen ([com.facebook.react.modules.statusbar.StatusBarModule.setTranslucent]).
+   *
+   *
+   * In case 'android:'windowTranslucentStatus' is used in activity's theme, it has to be removed in order to make RN's Status Bar API work.
+   * Out approach to achieve translucency of StatusBar has to be aligned with RN's approach to ensure [com.facebook.react.modules.statusbar.StatusBarModule] works.
+   *
+   *
+   * Links to follow in case of need of more detailed understating.
+   * https://chris.banes.dev/talks/2017/becoming-a-master-window-fitter-lon/
+   * https://www.youtube.com/watch?v=_mGDMVRO3iE
+   */
+  fun configureStatusBar(manifest: RawManifest, activity: Activity) {
+    val statusBarOptions = manifest.getAndroidStatusBarOptions()
+    val statusBarStyle = statusBarOptions?.optString(ExponentManifest.MANIFEST_STATUS_BAR_APPEARANCE)
+    val statusBarBackgroundColor = statusBarOptions?.optString(ExponentManifest.MANIFEST_STATUS_BAR_BACKGROUND_COLOR, null)
+
+    val statusBarHidden = statusBarOptions != null && statusBarOptions.optBoolean(
```

Comment:
To keep these as booleans it's a bit cleaner to keep the explicit null check. Otherwise they end up as `Boolean?` which can be a bit wonky to deal with.